jadx-gui/build.gradle.kts

```
@@ -110,11 +110,19 @@ project.components.withType(AdhocComponentWithVariants::class.java).forEach { c
 
 tasks.startShadowScripts {
 	doLast {
-		val newContent =
+		val newWindowsScriptContent =
 			windowsScript.readText()
 				.replace("java.exe", "javaw.exe")
 				.replace("\"%JAVA_EXE%\" %DEFAULT_JVM_OPTS%", "start \"jadx-gui\" /B \"%JAVA_EXE%\" %DEFAULT_JVM_OPTS%")
-		windowsScript.writeText(newContent)
+		// Add launch script path as a property
+		val newUnixScriptContent =
+			unixScript.readText()
+				.replace(
+					Regex("DEFAULT_JVM_OPTS=.+", RegexOption.MULTILINE),
+					{ result -> result.value + "\" \\\"-Djadx.launchScript.path=\$(readlink -e -- $0)\\\"\"" },
```

```suggestion
					{ result -> result.value + "\" \\\"-Djadx.launchScript.path=\$PRG\\\"\"" },
```
Looks like startup script already has full path to executable in `PRG` variable

---

Unfortunately it doesn't :(  
$PRG is only resolved if $0 is a symlink, and even then from my testing it will be resolved into a relative path if $0 is one, which means it is not suitable for us as we need the full path in the desktop file  
While checking this I also realized `readlink` is a part of GNU coreutils, so I swapped it out for the similar `realpath` command which is a part of POSIX